qir/qat/Passes/StaticResourceComponent/ReplaceQubitOnResetPass.cpp

```diff
@@ -198,12 +198,6 @@ llvm::PreservedAnalyses ReplaceQubitOnResetPass::run(llvm::Function& function, l
                     new_instr = new llvm::IntToPtrInst(new_index, pointer_type);
                     builder.Insert(new_instr);
 
-                    auto op_as_instr = llvm::dyn_cast<llvm::Instruction>(op);
-                    if (op_as_instr)
-                    {
-                        to_remove.push_back(op_as_instr);
-                    }
-
```

Comment:
@idavis clarified he was asking about this part of the code, so I'll explain more the motivation here...

This code added instances of the out-of-date LLVM instruction to the list of instructions to remove at the end of the pass. In that if-statement below, the instructions are only removed if they have no uses remaining, which the replacement here ensures. However, there is another bug somewhere in here that causes the same instruction to get added into the list more than once, which will cause a crash when the already removed instruction is removed a second time. As it turns out, these leftover instructions don't need to be pruned here because we run the dead code elimination pass after replace-on-reset, so it was simpler to delete the extraneous removal tracking.